lib/ansible/module_utils/network/eos/config/static_routes/static_routes.py

```diff
@@ -0,0 +1,376 @@
+            existing_static_routes_facts = self.get_static_routes_facts()
+        else:
+            existing_static_routes_facts = []
+
+        if self.state in self.ACTION_STATES or self.state == 'rendered':
+            commands.extend(self.set_config(existing_static_routes_facts))
+
+        if commands and self.state in self.ACTION_STATES:
+            if not self._module.check_mode:
+                for command in commands:
+                    self._connection.edit_config(command)
+            result['changed'] = True
+        if self.state in self.ACTION_STATES:
+            result['commands'] = commands
+        if self.state in self.ACTION_STATES or self.state == 'gathered':
+            changed_static_routes_facts = self.get_static_routes_facts()
+        elif self.state == 'rendered':
+            result['rendered'] = commands
+        elif self.state == 'parsed':
+            result['parsed'] = self.get_static_routes_facts(data=self._module.params['running_config'])
+        else:
+            changed_static_routes_facts = []
+
+        if self.state in self.ACTION_STATES:
+            result['before'] = existing_static_routes_facts
+            if result['changed']:
+                result['after'] = changed_static_routes_facts
+        elif self.state == 'gathered':
+            result['gathered'] = changed_static_routes_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_static_routes_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        onbox_configs = []
+        for h in existing_static_routes_facts:
+            return_command = add_commands(h)
+            for command in return_command:
+                onbox_configs.append(command)
+        config = self._module.params.get('config')
+        want = []
+        if config:
+            for w in config:
+                want.append(remove_empties(w))
+        have = existing_static_routes_facts
+        resp = self.set_state(want, have)
+        for want_config in resp:
+            if want_config not in onbox_configs:
+                commands.append(want_config)
+        return commands
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        if self.state in ('merged', 'replaced', 'overridden') and not want:
```

Comment:
We need to handle the situation where `config` key is supplied in the task, but it it empty. In this case, the `required_if` will pass, but will most definitely give a traceback.

Same goes for `state: parsed` and `running_config`.